### ISSUE TITLE
Horizontal thumbnails

### ISSUE THREAD

**Issue by Webdude737** [fancyapps/fancybox#1673]
For version 3, if you style the thumbnails to be horizontal like this:

https://codepen.io/fancyapps/pen/PjYPMw

but have a lot of images so there are multiple rows of images, when you activate the lightbox by clicking on a link, it correctly shows the image, and the corresponding thumbnail has blue borders BUT if the thumbnail is on a row other than the first row (hidden but you can get to it by going to the last slide in the row and using the up and down arrows), the thumbnail is not navigated to..

In other words, the image up top is fine, but the active slide is not scrolled to in the thumbnail list.  You have to click on  he left or right arrows to get it to slide to the correspong row that the thumbnail is in.

How I can fix this so that it slides to the currently active thumbnail upon activation of fancybox?  Thanks in advance.

**Comment by TheZoker** [comment]
Awesome! Could not agree more

**Comment by fancyapps** [comment]
Hi,

This has been improved in the last commit, see example - https://codepen.io/fancyapps/pen/RjRQzm

**Comment by TheZoker** [comment]
@fancyapps Could you maybe add a horizontal scroll, if there is not enough room for all thumbnails?

**Comment by fancyapps** [comment]
Hi,

There is (since the last commit) a new option `axis` to change scrolling direction, example:

```JS
$('[data-fancybox="images"]').fancybox({
  margin : [44,0,22,0],
  thumbs : {
    autoStart : true,
    axis : 'x'
  }
})
```
This is just for scrolling, you would still need to use CSS to change position.

See demo - https://codepen.io/anon/pen/xPrQEb

**Comment by TheZoker** [comment]
@fancyapps Short question:
I wanted to hide the thumbnails on mobile devices and added following line:
```javascript
onInit: function( instance, current ) {
     if($(window).width() < 800) {
          instance.Thumbs.hide();
     }
}
```
But I only get `Cannot read property 'hide' of undefined`. Why? It should work, right?

**Comment by fancyapps** [comment]
This is related to the order how events are processed. fancyBox fires several events and you can pass custom callback as an option to process any of them. And modules (like "thumbs") also attach their own event handlers. Your callback (for example, "onInit" in your sample) gets processed first, therefore there is no "Thumbs" instance created yet and you receive a warning.

So, if you want to disable thumbnails on some condtion, replace this 

`instance.Thumbs.hide();`

with:

`instance.group[instance.currIndex].opts.thumbs = false;`


About that `FullScreen.request` - it is just a wrapper for `Element.requestFullscreen()`, see https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API - you can request fullscreen mode on some element and `document.documentElement` is used by default.

**Comment by fancyapps** [comment]
btw, you could also use `mobile` option to automatically disable for mobile devices, for example, :

```JS
thumbs : {
    autoStart: true
},
mobile : {
    thumbs : false
}
```

**Comment by TheZoker** [comment]
@fancyapps Thank you for pointing that out! Very helpfull :)

Also, the mobile option is very nice!

Only one last thing: Is there an option to change the background opacity on mobile (I don't want transparent on mobile) without css?

**Comment by fancyapps** [comment]
You could change opacity using JS only, but then you would lost fading transition, because opacity changes by toggling CSS class name `fancybox-is-open` to the container.

You could, for example, add some custom class name to the container, for example:

```JS
onInit : function( instance ) {
        instance.$refs.container.addClass( 'fancybox-' + ($.fancybox.isMobile ? 'mobile' : 'desktop') );
}
```

and then use that to customize opacity level:

```CSS
.fancybox-mobile.fancybox-is-open .fancybox-bg {
    opacity: 1;
}
```

**Comment by TheZoker** [comment]
Thank you! I guess I use css then. Was just curious if I missed that option (like I missed the mobile option) :D But css is fine :)

Maybe you could add an option like:
```
opacity: '0.8',
mobile : {
    opacity: '1'
}
```
Only if css transition not get's lost of course

**Comment by fancyapps** [comment]
Maybe someday in the future. 

You see, to make it configurable through JS, it would be needed to set inline style. And then, if someone would want to change it using CSS, some ugly workaround would be required, (e.g, "!important" rule).

**Comment by TheZoker** [comment]
Yes that's not very smooth. Then just drop the idea 😄
